### DISCUSSION TITLE
Current Status of WASM support

### DISCUSSION THREAD

**Discussion by ChristophP** [parcel-bundler/parcel#9656] [Q&A] [answered]
I know Parcel v1 supported importing .wasm assets.
https://en.parceljs.org/webAssembly.html

For V2 I couldn't find anything in the docs about it. There are some issues that talked about it but it's unclear whether this exists as of now. There a [note in the Roadmap](https://github.com/parcel-bundler/parcel/discussions/7345#wasm) and also [this open issue](https://github.com/parcel-bundler/parcel/issues/1325).

So my question is: What is the current state WASM support in Parcel V2? Is there any support WASM at all currently? Could wasm be used by simply fetching and loading it with normal WASM APIs?

**Comment by mischnic** [comment, marked as answer]
> Could wasm be used by simply fetching and loading it with normal WASM APIs?

Yes, this would work:
```js
let wasm = fetch(new URL("./simple.wasm", import.meta.url))
WebAssembly.instantiateStreaming(wasm)
```

That Wasm file can be treated like any other asset: https://parceljs.org/languages/javascript/#url-dependencies


> Is there any support WASM at all currently?

You currently cannot ESM-`import` Wasm files, that would also be blocked on TLA support in Parcel: https://github.com/parcel-bundler/parcel/issues/4028
But this shouldn't prevent you from using Wasm

**Reply by ChristophP** [reply]
Great, will try the URL dependencies approach. That should do for my purposes. 👍

Out of curiosity, is having TLA blocked by something major or does it require bigger under the hood changes somehow to implement it? What's the main challenge?

**Reply by mischnic** [reply]
> does it require bigger under the hood changes somehow to implement it

Yes

https://github.com/orgs/web-infra-dev/discussions/9

**Reply by ChristophP** [reply]
I see. Thx for your answers.
